src/services/counters/CounterService.ts

```diff
@@ -0,0 +1,176 @@
+    let counterAddress: string [] = this.getCountersConfigurationAddress(cacheContainer, profile);
+    let request: IDmrRequest = {
+      address: counterAddress
+    };
+    this.dmrService.readResource(request).then(r => {
+      deferred.resolve(true);
+    }).catch(e => {
+      deferred.resolve(false);
+    });
+    return deferred.promise;
+  }
+
+  createCountersDMRPath(cacheContainer: string, profile: string): ng.IPromise<boolean> {
+    let deferred: ng.IDeferred<boolean> = this.$q.defer<boolean>();
+    let counterAddress: string [] = this.getCountersConfigurationAddress(cacheContainer, profile);
+    let request: IDmrRequest = {
+      address: counterAddress
+    };
+    this.dmrService.add(request).then(r => {
+      deferred.resolve(true);
+    }).catch(e => {
+      deferred.resolve(false);
+    });
+    return deferred.promise;
+  }
+
+  getAllCounters(cacheContainer: ICacheContainer): ng.IPromise<ICounter[]> {
+
+    let deferred: ng.IDeferred<ICounter[]> = this.$q.defer<ICounter[]>();
+    this.jgroupsService.getServerGroupCoordinator(cacheContainer.serverGroup).then((coord: IServerAddress) => {
+
+      let request: IDmrRequest = <IDmrRequest>{
+        address: this.getCountersRuntimeAddress(cacheContainer.name, coord.host, coord.name),
+        "recursive-depth": 2,
+        "include-runtime": true
+      };
+
+      this.dmrService.readResource(request)
+        .then((response: any): ICounter[] => {
+          let counters: ICounter[] = [];
+          let trail: String [] = [];
+          traverse(response, (key: string, value: string, trail: string []) => {
+            let traversingCounterObject: boolean = (key === "name");
+            if (traversingCounterObject) {
+              let counter: any = deepValue(response, trail);
+              let type: string = trail[0];
+              if (isNotNullOrUndefined(counter) && type === "strong-counter") {
+                counters.push(this.createStrongCounter(counter.name, counter.storage, counter["initial-value"],
+                  counter.value, counter["lower-bound"], counter["upper-bound"], counter));
+              } else if (isNotNullOrUndefined(counter) && type === "weak-counter") {
+                counters.push(this.createWeakCounter(counter.name, counter.storage, counter["initial-value"],
+                  counter.value, counter.concurrency, counter));
+              }
+            }
+          }, trail);
+          return counters;
+        }).then(counters => {
+        deferred.resolve(counters);
+      })
+    });
+    return deferred.promise;
+  }
+
+  update(counter: ICounter, profile: string, container: string): ng.IPromise<any> {
+    return this.save(counter, profile, container, ["is-new-node"]);
+  }
+
+  create(counter: ICounter, profile: string, container: string): ng.IPromise<any> {
+    counter["is-new-node"] = true;
+    return this.doCountersExist(container, profile).then((countersExists: boolean) => {
```

Comment:
this can be simplified to `then(countersExist => ...`

Reply:
We need to properly handle the failure of any of these promises, that way an error message will be displayed on the modal